real_preprocessing/src/camera_pose.cpp

```diff
@@ -0,0 +1,263 @@
+      cv::Mat rodrigues;
+      cv::Rodrigues(rotation_mat,rodrigues);
+
+      fout << "\n - targetID: " << std::to_string(id[i]);
+      fout << "\n   world_T_target:";
+      fout << "\n    rotation: [ "+std::to_string(rodrigues.at<double>(0, 0))+" , "<< std::to_string(rodrigues.at<double>(0, 1)) +" , "+ std::to_string(rodrigues.at<double>(0, 2))+" ]";
+      fout << "\n    translation: [ "+std::to_string(tag_trans(0, 3))+" , "<< std::to_string(tag_trans(1, 3))+" , "+ std::to_string(tag_trans(2, 3))+" ]";
+      fout << "\n   obj_points_in_target:";
+      fout << "\n    0: [ " + std::to_string(-(size[i]/2)) + ", " + std::to_string(-(size[i]/2)) + ", "+ std::to_string(0) +" ]"; //bl
+      fout << "\n    1: [ " + std::to_string(size[i]/2) + ", " + std::to_string(-(size[i]/2)) + ", "+ std::to_string(0) +" ]";  //br   
+      fout << "\n    2: [ " + std::to_string(size[i]/2) + ", " + std::to_string(size[i]/2) + ", "+ std::to_string(0) +" ]";   //tr
+      fout << "\n    3: [ " + std::to_string(-(size[i]/2)) + ", " + std::to_string(size[i]/2) + ", "+ std::to_string(0) +" ]";  //tl   
+    }
+    fout.close();
+  }
+
+  //solvePnP calculator given file number and tag position in file
+  Eigen::MatrixXd objTcam(int loc, int filenum)
+  {
+    YAML::Node tags_pix = YAML::LoadFile("detections_" + std::to_string(filenum) + ".yaml");
+    int bl_pix_x = tags_pix["detections"][loc]["corners"]["0"][0].as<int>();
+    int bl_pix_y = tags_pix["detections"][loc]["corners"]["0"][1].as<int>();
+    int br_pix_x = tags_pix["detections"][loc]["corners"]["1"][0].as<int>();
+    int br_pix_y = tags_pix["detections"][loc]["corners"]["1"][1].as<int>();
+    int tr_pix_x = tags_pix["detections"][loc]["corners"]["2"][0].as<int>();
+    int tr_pix_y = tags_pix["detections"][loc]["corners"]["2"][1].as<int>();
+    int tl_pix_x = tags_pix["detections"][loc]["corners"]["3"][0].as<int>(); 
+    int tl_pix_y = tags_pix["detections"][loc]["corners"]["3"][1].as<int>();
+    int tag_id = tags_pix["detections"][loc]["targetID"].as<int>();
+    double tag_size = tags_pix["detections"][loc]["size"][0].as<double>();
+
+    std::vector<cv::Point2d> img_pts;
+    std::vector<cv::Point3d> obj_pts;
+    cv::Mat obj_T_cam_rvec;
+    cv::Mat obj_T_cam_tvec;
+    cv::Mat rodrigues_rvec;
+    
+    img_pts.push_back(cv::Point2d(bl_pix_x, bl_pix_y));
+    img_pts.push_back(cv::Point2d(br_pix_x, br_pix_y));
+    img_pts.push_back(cv::Point2d(tr_pix_x, tr_pix_y));
+    img_pts.push_back(cv::Point2d(tl_pix_x, tl_pix_y));
+
+    //object corner specification
+    obj_pts.push_back(cv::Point3d(-(tag_size / 2), -(tag_size / 2), 0));
+    obj_pts.push_back(cv::Point3d((tag_size / 2), -(tag_size / 2), 0));
+    obj_pts.push_back(cv::Point3d((tag_size / 2), (tag_size / 2), 0));
+    obj_pts.push_back(cv::Point3d(-(tag_size / 2), (tag_size / 2), 0));
+
+    cv::Mat cam_matrix(3,3,CV_64F);
+    cv::Vec<float, 5> distCoeffs;
+    intrinsicLoad(cam_matrix, distCoeffs);
```

Comment:
Nit: I see here that you load in constant values every time this function is called. You can just call intrinsicLoad in the constructor and save the parameters in constants according to the cpp ref guidelines.